### DISCUSSION TITLE
INSTANCE KEY in WOOK Received Message

### DISCUSSION THREAD

**Discussion by diemesolutions** [salman0ansari/whatsapp-api-nodejs#387] [Q&A] [answered]
In wook of message webhook don't exist this instance key.

It's possible change?

**Comment by salman0ansari** [comment]
what?

**Reply by came** [reply]
I think he just wants to receive the instance id (or key) within all webhook (SendWebhook) calls. When you have a single shared webhook for all initialized instances you don't know who received e.g. the message. 
As far as I can tell this could be easily added to the body of the SendWebhook function - here: 

https://github.com/salman0ansari/whatsapp-api-nodejs/blob/8c966e5aba683496ac64bc7570d5851ae49d58cc/src/api/class/instance.js#L61

**Reply by came** [reply]
well, looks like there is a key defined in the webhook, but only for incoming messages:
 
https://github.com/salman0ansari/whatsapp-api-nodejs/blob/6d0bb841434198648db8630f59e99272687df878/src/api/class/instance.js#L225

**Reply by came** [reply]
mmmh, looks like the key prop from instance.key gets overwritten by the msg.key coming from whatsapp.
This is the webhook payload for an incoming message. As there is no instance key in there you don't know who received the message. The only option I could think of would be having a unique webhook for every instance :/

![Webhook_site_-_Test__process_and_transform_emails_and_HTTP_requests](https://user-images.githubusercontent.com/149962/203410042-cfcf8921-87ac-4cd1-a31b-dfaa6787999f.jpg)

**Comment by salman0ansari** [comment, marked as answer]
fixed in the latest commit f90287260cd3395d2ea2234a3f61bdc6e513397f
